feat(many): add release notes and updates for 1.16 (#183)
anbox-cloud/discourse-docs: announcements/1.16.0.md

### Anbox runtime

#### Split of Android's system and vendor directories

With the introduction of support for Android 13, Anbox Cloud now separates the `/system` and `/vendor` partitions of Android into separate directories. Inside the LXD containers, the two partitions are available at `/var/lib/anbox/android-system` and `/var/lib/anbox/android-vendor`. From a user perspective nothing changes though, because both directories are combined at `/var/lib/anbox/rootfs`. Therefore, no changes to existing addons or customisations are required. The split allows to use custom Android system images in a future Anbox Cloud release.

#### New bug report utility

The Anbox [images](https://anbox-cloud.io/docs/ref/provided-images) include a new bug report tool called `anbox-bug-report`, which is extensible and collects information in an archive rather than a file. This method allows developers and support to easier inspect the information when debugging issues. See the [documentation](https://anbox-cloud.io/docs/howto/troubleshoot/bug-report) for more details.

### WebRTC streaming improvements

> To further allow better fine-tuning of the WebRTC based streaming implementation in Anbox Cloud, 1.16 includes various small improvements:

Reviewer: ```suggestion
To allow better fine-tuning of the WebRTC based streaming implementation in Anbox Cloud, 1.16 includes various small improvements:
```

Reviewer: Either "further" or "better", but not both. ;)

Reviewer 2: You're right, I misread your suggested changes!

* Video bitrate limits can now be specified on a per resolution/frame rate level.

> * Additional options allow enabling NVIDIA-specific options for H.264 encoding ([adaptive quantisation](https://docs.nvidia.com/video-technologies/video-codec-sdk/nvenc-video-encoder-api-prog-guide/#adaptive-quantization-aq), [multipass](https://docs.nvidia.com/video-technologies/video-codec-sdk/nvenc-video-encoder-api-prog-guide/#multi-pass-frame-phencoding)).

Reviewer: ```suggestion
* Additional options allow enabling NVIDIA-specific options for H.264 encoding ([adaptive quantisation](https://docs.nvidia.com/video-technologies/video-codec-sdk/nvenc-video-encoder-api-prog-guide/#adaptive-quantization-aq), [multi-pass frame encoding](https://docs.nvidia.com/video-technologies/video-codec-sdk/nvenc-video-encoder-api-prog-guide/#multi-pass-frame-phencoding)).
```

* Audio and video are now sent on separate streams to improve playout delay on the client.
* The benchmark utility can now provide statistics for minimum playout and jitter buffer delay.
* TURN credentials for the container are now dynamically allocated and are no longer created with a constant lifetime.

### Data channel v2

> [Version 2 of the OOB data exchange protocol](https://anbox-cloud.io/docs/howto/stream/oob-data#oob-v2), which Anbox Cloud introduced in version 1.15, provides a full-duplex bidirectional data transmission between the Android container and the WebRTC client. In the 1.16 release, Anbox Cloud introduced the Anbox WebRTC data proxy service as an extension to version 2 of the OOB data. The data proxy service exposes a binder service in the Android container that can be used by an Android application to communicate with a WebRTC client over the Anbox WebRTC platform.

Reviewer: ```suggestion
[Version 2 of the OOB data exchange protocol](https://anbox-cloud.io/docs/howto/stream/oob-data#oob-v2), which Anbox Cloud introduced in version 1.15, provides a full-duplex bidirectional data transmission between the Android container and the WebRTC client. In the 1.16 release, Anbox Cloud introduces the Anbox WebRTC data proxy service as an extension to version 2 of the OOB data. The data proxy service exposes a binder service in the Android container that can be used by an Android application to communicate with a WebRTC client over the Anbox WebRTC platform.
```

### Ubuntu 22.04 support

The Juju charms of Anbox Cloud now fully support Ubuntu 22.04. The Anbox Cloud Appliance can run on Ubuntu 22.04 already since the 1.15 release.

By default, the Anbox Cloud bundles ([`anbox-cloud`](https://charmhub.io/anbox-cloud), [`anbox-cloud-core`](https://charmhub.io/anbox-cloud-core)) available from [Charmhub](https://charmhub.io/) use Ubuntu 22.04.

With the introduction of Ubuntu 22.04, support for Ubuntu 18.04 was removed.

### AMS improvements

#### APT mirror configuration

The APT mirror used inside Anbox containers has been hard-coded to `archive.ubuntu.com` or `ports.ubuntu.com`, which is not always ideal. To allow for easier modification, AMS now provides a `container.apt_mirror` configuration option. This option can be set to a custom Ubuntu mirror at runtime and will then be used by all containers that are started afterwards.

#### Networking ACLs

> A new AMS charm configuration `use_network_acl` has been added. This configuration enables AMS to use [LXD network ACLs](https://linuxcontainers.org/lxd/docs/master/howto/network_acls/) <!-- wokeignore:rule=master --> rather than the AMS node controller (which is used by default) to restrict the network access of Anbox containers. The option is experimental and not ready for general use, but will in the future replace the functionality that is currently part of the AMS node controller.

Reviewer: ```suggestion
A new AMS charm configuration `use_network_acl` has been added. This configuration enables AMS to use [LXD network ACLs](https://linuxcontainers.org/lxd/docs/latest/howto/network_acls/) rather than the AMS node controller (which is used by default) to restrict the network access of Anbox containers. The option is experimental and not ready for general use, but will in the future replace the functionality that is currently part of the AMS node controller.
```

### Other

* Included Android security updates for November 2022 (see [Android Security Bulletin - November 2022](https://source.android.com/docs/security/bulletin/2022-11-01) for more information).
* Updated Android WebView to [107.0.5304.105](https://chromereleases.googleblog.com/2022/11/chrome-for-android-update.html).

> * The used NVIDIA driver series is updated to [515](https://docs.nvidia.com/datacenter/tesla/index.html) which is supported until May 2023.

Reviewer: ```suggestion
* Updated the used NVIDIA driver series to [515](https://docs.nvidia.com/datacenter/tesla/index.html), which is supported until May 2023.
```

> * The appliance now deploy Ubuntu 22.04 based LXD containers on new deployments. Existing deployments will keep their existing Ubuntu 20.04 ones.

Reviewer: ```suggestion
* Updated the appliance to deploy LXD containers based on Ubuntu 22.04 on new deployments (existing deployments keep their existing containers based on Ubuntu 20.04).
```

> * AMD GPU support is now automatically detected by the [`ams-lxd`](https://charmhub.io/ams-lxd) Juju charm and the Anbox Cloud Appliance and the kernel `amdgpu` driver configured for best performance and support.

Reviewer: ```suggestion
* Updated the [`ams-lxd`](https://charmhub.io/ams-lxd) Juju charm and the Anbox Cloud Appliance to automatically detect AMD GPU support and configure the kernel `amdgpu` driver for best performance and support.
```
Is that correct? The sentence was quite ambiguous.

Author: Yes, after your change it's correct.

> * The [AWS g4ad instance types](https://aws.amazon.com/ec2/instance-types/g4/) (equipped with AMD v520 GPUs) are now supported by the [Anbox Cloud Appliance images](https://aws.amazon.com/marketplace/pp/prodview-3lx6xyaapstz4) available through the AWS marketplace.

Reviewer: ```suggestion
* Added support for the [AWS `g4ad` instance types](https://aws.amazon.com/ec2/instance-types/g4/) (equipped with AMD v520 GPUs) to the [Anbox Cloud Appliance images](https://aws.amazon.com/marketplace/pp/prodview-3lx6xyaapstz4) that are available through the AWS marketplace.
```

> * The appliance can now be initialized as any Ubuntu user, not just the `ubuntu` one.

Reviewer: ```suggestion
* Updated the appliance so that it can be initialised as any Ubuntu user, not just the `ubuntu` one.
```
